### PR TITLE
[New Term] C++ Math Functions: `atanh()`

### LINKED ISSUE #955: [Term Entry] C++ Math Functions: `atanh()`
### Reason/inspiration (optional)

Under `cpp/concepts/math-functions/terms/`, create a new term entry for the `atanh()` function.

- In the `terms/` folder, make a new folder called `atanh`
- Inside the `atanh` folder, create a new markdown file called `atanh.md`
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Let's include an "Example" section with a \```cpp code snippet.
- Lastly, let's include a "Codebyte Example" with a runnable \```codebyte/cpp

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by sumedhav2004**
## Description
 
Added new term entry C++ Math Functions: atanh()

Closes #955

 ## Type of Change

 Adding a new entry
 
 ## Checklist
 * [x]   All writings are my own.
 * [x]    My entry follows the Codecademy Docs style guide.
 * [x]    My changes generate no new warnings.
 * [x]    I have performed a self-review of my own writing and code.
 * [x]    I have checked my entry and corrected any misspellings.
 * [x]    I have made corresponding changes to the documentation if needed.
 * [x]    I have confirmed my changes are not being pushed from my forked main branch.
 * [x]    I have confirmed that I'm pushing from a new branch named after the changes I'm making.
 * [x]    Under "Development" on the right, I have linked any issues that are relevant to this PR.

### CODE REVIEW

File: content/cpp/concepts/math-functions/terms/atanh/atanh.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Functions'
+  - 'Arithmetic'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+The **`atanh()`** function in C++ returns the arc hyperbolic tangent (inverse hyperbolic tangent) of a number in `radians`.The area hyperbolic tangent is the inverse operation of the hyperbolic tangent.
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atanh/atanh.md:14]:
```suggestion
The **`atanh()`** function in C++ returns the arc hyperbolic tangent (inverse hyperbolic tangent) of a number in radians.
```

File: content/cpp/concepts/math-functions/terms/atanh/atanh.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+## Syntax
+
+```pseudo
+atanh(data_type x)
+```
+
+`x`: The value of mandatory argument in range `[-1,1]`.
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atanh/atanh.md:22]:
```suggestion
`x` is the value of mandatory argument in range `[-1,1]`.
```

File: content/cpp/concepts/math-functions/terms/atanh/atanh.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+## Parameter
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atanh/atanh.md:24]: Remove the parameter header and place this sentence at the end of the "Syntax" section.

File: content/cpp/concepts/math-functions/terms/atanh/atanh.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	cout << "atanh(x) = " << result << " radian" << endl;
+
+	// result in degrees
+	cout << "atanh(x) = " << result*180/PI << " degree" << endl;
+	
+	return 0;
+}
+```
+
+Output:
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atanh/atanh.md:74]:
```suggestion
This example produces the following output:

```

File: content/cpp/concepts/math-functions/terms/atanh/atanh.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+If the argument is not on the interval `[-1, +1]`, a range error occurs.
+
+If the argument is `±1`, a pole error occurs.
+
+If the argument is `±0`, it is returned `unmodified`.
+If the argument is `±1`, `±∞` is returned and `FE_DIVBYZERO` is raised.
+If |arg|>1, `NaN` is returned and `FE_INVALID` is raised.
+If the argument is `NaN`, `NaN` is returned.
+
+## Prototype [C++11 Standard]
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atanh/atanh.md:39]: I don't think we need this section. It's at odds with all the other math entries.

File: content/cpp/concepts/math-functions/terms/atanh/atanh.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+atanh(data_type x)
+```
+
+`x`: The value of mandatory argument in range `[-1,1]`.
+
+## Parameter
+
+This function accepts one mandatory parameter `x` which specifies the inverse hyperbolic angle in radian which lies in the range `[-1, 1]`. The parameter can be of `double`, `float`, or `long double` datatype.
+
+## Error Handling
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atanh/atanh.md:28]: I'd remove the header and include this under "Syntax"

File: content/cpp/concepts/math-functions/terms/atanh/atanh.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+If the argument is not on the interval `[-1, +1]`, a range error occurs.
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/atanh/atanh.md:30]: I'd take all of these following lines and make them a bulleted list.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@sumedhav2004, I did a first review and left a few comments.

**Comment by SSwiniarski** [comment]
@sumedhav2004, just checking up if you have any issues addressing the above comments. Let us know if you need any help.

**Comment by yangc95** [comment]
> @sumedhav2004, just checking up if you have any issues addressing the above comments. Let us know if you need any help.

_Direct changes will be made in T-1 week if there are no further communications._

**Comment by SSwiniarski** [comment]
@sumedhav2004 in the interest of getting this merged, I went ahead and applied the above changes.

@Dusch4593, @yangc95 anything else that needs to be edited before we merge this?

**Review by Dusch4593 — APPROVED** [review]
cc: @sumedhav2004 @SSwiniarski @yangc95 

LGTM as well! 👍🏻

I made the following edits in ec6265212fb9642fffea6a11eb027e03badbb04b that I wanted to note here:

- Removed reference to "arc hyperbolic tangent" because [it's noted in other docs sites](https://en.cppreference.com/w/cpp/numeric/math/atanh#:~:text=Although%20the%20C,area%20hyperbolic%20tangent%22.) that these kinds of functions refer to the _area_, not the _arc_.
- Rephrasing on line # 22, including removal of backticks from range and insertion of link to C++ Data Types entry.
- Separated out last four list items to point out they refer to C++ versions that support IEEE floating-point arithmetic.
- Rephrase on lines # 36 and # 67.

**Comment by Dusch4593** [comment]
Congrats and thank you for your contribution to Codecademy Docs, @sumedhav2004! 

The entry should now be live: https://www.codecademy.com/resources/docs/cpp/math-functions/atanh

Your Codecademy profile is currently hidden from view on the entry page. To make it visible, you'll need to add your GitHub username to your Codecademy profile, log out of Codecademy, and log back in through your GitHub account:

Screen Shot 2022-12-03 at 8 07 17 AM

Looking forward to your next Docs contribution! 😄

**Pull request merged**